### PR TITLE
Update version and change log for 0.12.0 candidate release

### PR DESCRIPTION
**Pull request by cderici**
## Description

Add rc0.12.0 content to the change log, and update the versions in the docs.

### CODE REVIEW

File: templates/index.md.tmpl [hunk 1]
```diff
@@ -33,7 +33,7 @@ Work is ongoing to include support for more of the juju CLIs capabilities within
 
 ## Prerequisites
 
-* [Juju][0] `2.9.33+`
+* [Juju][0] `3.4+`
```
Comment (hmlanigan) [inline templates/index.md.tmpl:37]: this should be 2.9.49

File: docs/index.md [hunk 2]
```diff
@@ -90,7 +90,7 @@ Terraform 0.13 and later:
 terraform {
   required_providers {
     juju = {
-      version = "~> 0.11.0"
+      version = "~> 0.12.0"
```
Comment (hmlanigan) [inline docs/index.md:93]: this is a generated file, so the actual changes are somewhere else.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,24 @@
+## 0.12.0 (April 22, 2024)
+
+NOTES:
+* **This release requires juju controller version 3.4 or later juju.**
```
Comment (hmlanigan) [inline CHANGELOG.md:4]: It requires 2.9.49 or later for general content. To use with JAAS login enhancements, requires 3.5.0+. To use with juju secrets requires 3.4.0+

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+* This release uses juju client api code from the juju 3.5-beta1 candidate release.
+
+ENHANCEMENTS:
+* Schema and methods for user secrets management are implemented.
+* `resource_secret` and `resource_access_secret` resources are added.
+* Add grant and remove secret access function to `internal/juju`.
+* Data source for secrets is added.
```
Comment (hmlanigan) [inline CHANGELOG.md:11]: Let's combine these into a description of user secrets.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,24 @@
...
+* Client ID and secret are added to the provider config.
+* Update, and remove run limitations for build and unit test workflows.
+* README.md and GitHub issue templates are updated.
+* github.com/juju/names/v4 is updated to v5.
```
Comment (hmlanigan) [inline CHANGELOG.md:15]: These shouldn't be included as enhancements.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,24 @@
...
+
+BUG FIXES:
+
+* Validate channel in application resource by @Aflynn50 in https://github.com/juju/terraform-provider-juju/pull/447
```
Comment (hmlanigan) [inline CHANGELOG.md:19]: We should call this out more - due to the change in channel for charms requiring both track and risk now.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,24 @@
...
+NOTES:
+* **This release requires juju controller version 3.4 or later juju.**
+* This release uses juju client api code from the juju 3.5-beta1 candidate release.
+
+ENHANCEMENTS:
+* Schema and methods for user secrets management are implemented.
+* `resource_secret` and `resource_access_secret` resources are added.
+* Add grant and remove secret access function to `internal/juju`.
+* Data source for secrets is added.
+* Client ID and secret are added to the provider config.
```
Comment (hmlanigan) [inline CHANGELOG.md:12]: This one needs an enhanced description as well.

File: docs/index.md (hunk 2 above)
Comment (cderici) [reply docs/index.md:93]: yeah I didn't edit this file by hand, go generate made these changes based on my changes on the examples and the index template.

**Review by hmlanigan — APPROVED** [review]
Thanks for the updates. More eyes can be better with these change logs.

**Pull request merged**